main.tf

```diff
@@ -0,0 +1,12 @@
+
+resource "null_resource" "ansible_provisioner" {
+
+  provisioner "local-exec" {
+    command = "ansible-playbook ${var.playbook} -e host=${var.ip}"
```

Comment:
my concern is the scheduling within terraform. Terraform will run `local-exec` provisioners almost immediately. So for example, suppose there are 25 hosts, and a default of 10 concurrent operations. Then terraform will be blocked, since it's waiting on 10 local-exec provisioners to complete. Let's give TF the knowledge it needs to efficiently schedule executions.